hil/client/base.py

```diff
@@ -79,7 +80,8 @@ def _find_reserved(string, slashes_ok=False):
         p = r"[^A-Za-z0-9 /$_.+!*'(),-]+"
     else:
         p = r"[^A-Za-z0-9 $_.+!*'(),-]+"
-    return list(x for l in re.findall(p, string) for x in l)
+    # return all unique chars that exist neither in p nor string
+    return list(set(x for l in re.findall(p, string) for x in l))
```

Comment:
It took me >1 min to figure out how this worked (I think partially just because there was a lot of seeking involved in parsing the comprehension). What about:

1. Change the patterns to remove the `+`
2. `list(set(re.findall(p, string)))`

Also, I don't like the fact that we have two nearly identical regexes. One alternative:

```python
p = # regex without the /
result = # expression with re.findall
if not slashes_ok and '/' in string:
    result.append('/')

Reply:
@zenhack my code now should reflect your comments.  My logic is slightly different from what you recommended, but works just as it did before.